stl/inc/optional

```diff
@@ -952,8 +952,11 @@ _NODISCARD constexpr bool operator>=(const _Ty1& _Left, const optional<_Ty2>& _R
 }
 
 #if _HAS_CXX20
+template <class _Ty>
+concept _Derived_from_optional = requires(const _Ty& _Obj) { []<class _Uty>(const optional<_Uty>&) {}(_Obj); };
+
 _EXPORT_STD template <class _Ty1, class _Ty2>
-    requires (!_Derived_from_specialization_of<_Ty2, optional>) && three_way_comparable_with<_Ty1, _Ty2>
+    requires (!_Derived_from_optional<_Ty2>) && three_way_comparable_with<_Ty1, _Ty2>

```

Comment:
Looks like that I need to revert some changes and report related compiler bugs...